test/Microsoft.AspNetCore.Hosting.Tests/WebHostBuilderTests.cs

```diff
@@ -353,6 +352,41 @@ public void UseEnvironmentIsNotOverriden()
         }
 
         [Fact]
+        public void EnvironmentConfigurationFallbackHierarchyHonored()
+        {
+            var testServer = new TestServer();
+
+            // Default to Production
+            var host = new WebHostBuilder()
+                .UseServer(testServer)
+                .UseStartup("Microsoft.AspNetCore.Hosting.Tests")
+                .Build();
+
+            Assert.Equal(EnvironmentName.Production, host.Services.GetService<IHostingEnvironment>().EnvironmentName);
+
+            // Use environment variables if available
+            Environment.SetEnvironmentVariable(WebHostDefaults.EnvironmentVariablesPrefix + WebHostDefaults.EnvironmentKey, "EnvironmentA");
```

Comment:
Scary, let's *not* have a test for this. Global state like this affects tests running in parallel.